nav2_costmap_2d/include/nav2_costmap_2d/layered_costmap.hpp

```diff
@@ -135,6 +135,14 @@ class LayeredCostmap
     return &plugins_;
   }
 
+  /**
+   * @brief Get the vector of pointers to the costmap filters
+   */
+  std::vector<std::shared_ptr<Layer>> * getFilters()
```

Comment:
There is nothing wrong with this, since this is how the plugins are as well, but its a little odd to me that this is a pointer and not just a reference. It would make the semantics later in the code much easier if it was returned by reference so that it was just `filter->do_thing()` vs `(*filter)->do_thing()`.

Reply:
OK. In my opinion that is also suitable to go structure. The `plugins_` or `filters_` is a vector of shared pointers, so its iterator will have a double-pointer dimension (because of a `std::iterator` is a pointer to a vector' item which is a shared_ptr itself). Changing the `getFilters()` return value to a reference to std::vector will simplify the iterator boundaries in for-cycle (e.g. `filters->begin()` will be simplified to `filters.begin()`), but leave the `(*filter)->do_thing` odd construction to be the same. So I see no point in replacing pointer to a reference return value.